c_plus_plus/src/model/recorder/recorder.cpp

```diff
@@ -0,0 +1,86 @@
+#include "recorder.h"
+
+#include <QCamera>
+#include <QCameraInfo>
+#include <QCameraViewfinder>
+#include <QMediaRecorder>
+#include <QUrl>
+
+namespace Model
+{
+
+Recorder::Recorder(const QString cameraDevice, QWidget *parent)
+    : IRecorder(parent)
+    , m_camera(new QCamera(getCameraInfo(cameraDevice), this))
+    , m_mediaRecorder(new QMediaRecorder(m_camera, this))
+{
+    m_camera->setCaptureMode(QCamera::CaptureMode::CaptureVideo);
+
+    QVideoEncoderSettings videoSettings;
+    videoSettings.setQuality(QMultimedia::VeryHighQuality);
+    videoSettings.setCodec("video/x-vp8");
+    m_mediaRecorder->setVideoSettings(videoSettings);
+
+    QAudioEncoderSettings audioSettings;
+    audioSettings.setQuality(QMultimedia::VeryHighQuality);
+    audioSettings.setCodec("audio/x-vorbis");
+    m_mediaRecorder->setAudioSettings(audioSettings);
+
+    m_mediaRecorder->setContainerFormat("video/webm");
+}
+
+void Recorder::start(const QString outputPath)
+{
+    m_mediaRecorder->setOutputLocation(QUrl::fromLocalFile(outputPath + "/media"));
+
+    if(m_mediaRecorder->isAvailable() && m_camera->status() == QCamera::Status::ActiveStatus)
+    {
+        m_mediaRecorder->record();
+    }
+}
+
+void Recorder::stop()
+{
+    m_mediaRecorder->stop();
+}
+
+QCameraInfo Recorder::getCameraInfo(QString cameraDevice)
+{
+    QCameraInfo defaultCameraInfo = QCameraInfo::defaultCamera();
+
+    if(cameraDevice != "")
+    {
+        QList<QCameraInfo> cameras = QCameraInfo::availableCameras();
+
+        foreach (const QCameraInfo &cameraInfo, cameras)
```

Comment:
As you wish, but you can also write foreach loop like this:
```
for(auto cameraInfo : cameras)
```

Reply:
I will keep it that way, as they did in the Qt documentation